File: app/src/main/java/moe/yuuta/mipushtester/push/MessageDetailActivity.java
package moe.yuuta.mipushtester.push;

import android.os.Bundle;
import android.view.MenuItem;

import com.xiaomi.mipush.sdk.MiPushMessage;
import com.xiaomi.mipush.sdk.PushMessageHelper;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import moe.yuuta.mipushtester.R;
import moe.yuuta.mipushtester.databinding.ActivityMessageDetailBinding;

public class MessageDetailActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MiPushMessage message = (MiPushMessage) getIntent().getSerializableExtra(PushMessageHelper.KEY_MESSAGE);
        ActivityMessageDetailBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_message_detail);
        binding.setMessage(message);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/push/MessageDetailActivity.kt
package moe.yuuta.mipushtester.push

import android.os.Bundle
import android.view.MenuItem
import androidx.annotation.Nullable
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.xiaomi.mipush.sdk.MiPushMessage
import com.xiaomi.mipush.sdk.PushMessageHelper
import moe.yuuta.mipushtester.R
import moe.yuuta.mipushtester.databinding.ActivityMessageDetailBinding

class MessageDetailActivity : AppCompatActivity() {
    override fun onCreate(@Nullable savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val message: MiPushMessage = intent.getSerializableExtra(PushMessageHelper.KEY_MESSAGE) as MiPushMessage
        val binding: ActivityMessageDetailBinding = DataBindingUtil.setContentView(this, R.layout.activity_message_detail)
        binding.message = message
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean =
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            else -> false
        }
}
